### PR TITLE
Add deprecation notices to helm chart

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>

#### What this PR does

Adds a section for deprecation notices and migration instructions to the changelog. Mentions this section in NOTEX.txt (printed on helm upgrade/helm install) and README.md.

Also updates RLEASE.md with instructions to keep the deprecation notices up to date.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- NA Tests updated
- [x] Documentation added
- NA `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
 # Changelog
 
+## Deprecated features
+
+This section contains deprecated features and interfaces that the chart exposes. The deprecation policy of the chart is to
+remove a deprecated item on the third major release after it has been deprecated it.
```
Comment (osg-grafana) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:6]:
```suggestion
This section contains deprecated features and interfaces that the chart exposes. The deprecation policy of the chart is to
remove a deprecated item from the third major release after it has been deprecated it.
```

The preposition "on" is ambiguous here.

File: operations/helm/charts/mimir-distributed/README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ Helm chart for deploying [Grafana Mimir](https://grafana.com/docs/mimir/v2.2.x/)
 
 See the [Grafana Mimir version 2.2 release notes](https://grafana.com/docs/mimir/v2.2.x/release-notes/v2.2/) and [Upgrade the Grafana Mimir Helm chart from version 2.1 to 3.0](https://grafana.com/docs/mimir/latest/operators-guide/deploying-grafana-mimir/upgrade-helm-chart-2.1-to-3.0/).
 
-**IMPORTANT**: Always consult the [CHANGELOG.md](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/CHANGELOG.md) file to learn about breaking changes that require action during upgrade.
+**IMPORTANT**: Always consult the Deprecated features section in [CHANGELOG.md](./CHANGELOG.md#deprecated-features) file to learn about breaking changes that require action during upgrade.
```
Comment (osg-grafana) [inline operations/helm/charts/mimir-distributed/README.md:7]:
```suggestion
**IMPORTANT**: Always consult the Deprecated features section in the [CHANGELOG.md](./CHANGELOG.md#deprecated-features) file to learn about breaking changes that require action during upgrade.
```
Nit about missing definite article. Fix throughout.

**Review by osg-grafana — APPROVED** [review]

**Comment by dimitarvdimitrov** [comment]
will wait a day or two if someone from engineering wants to take a look and will merge after

**Review by krajorama — CHANGES_REQUESTED** [review]

File: operations/helm/charts/mimir-distributed/README.md.gotmpl [hunk 3]
```diff
@@ -4,7 +4,7 @@ Helm chart for deploying [Grafana Mimir]({{ template "chart.homepage" . }}) or o
 
 See the [Grafana Mimir version 2.2 release notes](https://grafana.com/docs/mimir/v2.2.x/release-notes/v2.2/) and [Upgrade the Grafana Mimir Helm chart from version 2.1 to 3.0](https://grafana.com/docs/mimir/latest/operators-guide/deploying-grafana-mimir/upgrade-helm-chart-2.1-to-3.0/).
 
-**IMPORTANT**: Always consult the [CHANGELOG.md](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/CHANGELOG.md) file to learn about breaking changes that require action during upgrade.
+**IMPORTANT**: Always consult the Deprecated features section in the [CHANGELOG.md](./CHANGELOG.md#deprecated-features) file to learn about breaking changes that require action during upgrade.
```
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/README.md.gotmpl]: I pushed a change in https://github.com/grafana/mimir/pull/2934/commits/7f69205a63ae029b3529935db891a2224b7465d2

File: operations/helm/charts/mimir-distributed/templates/NOTES.txt [hunk 4]
```diff
@@ -32,4 +32,4 @@ Ingress is not enabled, see the {{ if $.Values.enterprise.enabled }}gateway{{ el
 From inside the cluster:
   http://{{ include "mimir.fullname" $ }}-{{ if .Values.enterprise.enabled }}gateway{{ else }}nginx{{ end }}.{{ .Release.Namespace }}.svc:{{ $gateway.service.port }}{{ template "mimir.prometheusHttpPrefix" $ }}
 
-**IMPORTANT**: Always consult the CHANGELOG.md at https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/CHANGELOG.md file to learn about breaking changes that require action during upgrade.
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/NOTES.txt:35]: I just ran `helm upgrade` and a similar message showed up. It made me think whether or not this should be there. I already did the upgrade, isn't it too late to look at the changelog? 😄 

@krajorama

Comment (krajorama) [reply operations/helm/charts/mimir-distributed/templates/NOTES.txt:35]: It's a good reminder anyway I think

**Review by krajorama — APPROVED** [review]

**Pull request merged**